sopel/config/core_section.py

```diff
@@ -1048,6 +1048,16 @@ def homedir(self):
     silently from the triggering IRC user's perspective.
     """
 
+    sasl_cert_file = FilenameAttribute('sasl_cert_file')
```

Comment:
Setting name is a good suggestion. Just be glad I got away from the original idea of shoehorning this setting into the existing auth stuff… 😝 (I also hate how some settings have different meanings depending on which auth method is in use, and 8.0 might be a time to fix.)

If I understand your last line correctly, I think Libera does. Their instructions say to enable SASL EXTERNAL if connecting via Tor (because SASL is required over Tor, no matter what), but otherwise it seems to just auth against the cert regardless of whether SASL negotiation happened.

Reply:
"Directly verify client certs" meaning that the IRCd itself ensures a client cert is signed by a CA it trusts, and if not, it disconnects. The `cafile` parameter in [inspircd3 ssl_gnutls.so](https://docs.inspircd.org/3/modules/ssl_gnutls/) implies that's supported. In these cases, SASL isn't necessarily involved at all, so the `sasl_` config name may not have made sense.

Reply:
Fair. I think one of the best ways to find out if any networks validate client certs against CA roots is to ship this feature and move on to other projects, then wait for bug reports about it not working in that case. 😏